From db1fe1f6ac1c6f2841d3c3828f371b11df70d8e9 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Mon, 19 May 2014 19:26:19 +0200
Subject: [PATCH 2/2] Terminate with proper exit status upon error

$ meld x.txt y/
Usage:
  meld                              Start with an empty window
  meld <file|folder>                Start a version control comparison
  meld <file> <file> [<file>]       Start a 2- or 3-way file comparison
  meld <folder> <folder> [<folder>] Start a 2- or 3-way folder comparison

Traceback (most recent call last):
  File "/home/lkundrak/src/meld/meld/meldapp.py", line 70, in do_command_line
    tab = self.parse_args(command_line)
  File "/home/lkundrak/src/meld/meld/meldapp.py", line 291, in parse_args
    return tab if len(comparisons) == 1 else None
UnboundLocalError: local variable 'tab' referenced before assignment

After:

$ meld x.txt y/
Usage:
  meld                              Start with an empty window
  meld <file|folder>                Start a version control comparison
  meld <file> <file> [<file>]       Start a 2- or 3-way file comparison
  meld <folder> <folder> [<folder>] Start a 2- or 3-way folder comparison
---
 meld/meldapp.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meld/meldapp.py b/meld/meldapp.py
index 5418085..70d5ca7 100644
--- a/meld/meldapp.py
+++ b/meld/meldapp.py
@@ -282,6 +282,8 @@ class MeldApp(Gtk.Application):
                 parser.error(error)
             else:
                 print(error)
+            self.quit()
+            return 1
 
         return tab if len(comparisons) == 1 else None
 
-- 
1.8.3.1

